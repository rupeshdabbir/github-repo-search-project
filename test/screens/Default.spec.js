import React from 'react';
import renderer from 'react-test-renderer';
import {
  StyleSheetTestUtils,
} from 'aphrodite';
import {Provider} from 'react-redux';
import {
  StaticRouter as Router,
} from 'react-router-dom';
import configureMockStore from 'redux-mock-store';
import DefaultLayout, {
  pushUrlQuery,
} from 'screens/Default';

function createComponent(component) {
  const mockStore = configureMockStore();
  const store = mockStore();
  return renderer.create(
    <Provider store={store}>
      <Router context={{}}>
        {component}
      </Router>
    </Provider>
  ).toJSON();
}

describe('Screen: DefaultLayout', () => {

  beforeAll(() => {
    StyleSheetTestUtils.suppressStyleInjection();
  });

  afterAll(() => {
    StyleSheetTestUtils.clearBufferAndResumeStyleInjection();
  });

  describe('when passed a component', () => {
    it('should render it in the default layout', () => {
      const component = () => <p>testing</p>;
      const screen = <DefaultLayout component={component} />;
      expect(createComponent(screen)).toMatchSnapshot();
    });
  });

  describe('pushUrlQuery function', () => {
    it('should push a search query on the URL', () => {
      const spy = jest.fn();
      pushUrlQuery(spy, 'testing');
      expect(spy.mock.calls[0]).toMatchSnapshot();
    });
  });

});
