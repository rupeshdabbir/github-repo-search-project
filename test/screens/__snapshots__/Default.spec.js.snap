// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Screen: DefaultLayout pushUrlQuery function should push a search query on the URL 1`] = `
Array [
  Object {
    "pathname": "/search",
    "search": "per_page=42&page=1&q=undefined+language:undefined",
  },
]
`;

exports[`Screen: DefaultLayout when passed a component should render it in the default layout 1`] = `
<div
  className="DefaultLayout_1jug6qr u-flex u-flexCol"
>
  <div
    className="Container_16pg94n-o_O-HeaderContainer_sb4ugp"
  >
    <div
      className="Container_inner_w9z6jj-o_O-Container_padding_1cv73u5"
    >
      <div
        className="Header_1mnqmvf u-sm-flex u-sm-flexJustifyBetween u-sm-flexAlignItemsCenter"
      >
        <div
          className="u-sm-flexGrow2"
        >
          <div>
            <h1
              className="Logo_title_229208"
            >
              <a
                className="Logo_link_18deiwy u-flex"
                href="/"
                onClick={[Function]}
              >
                <svg />
                GitHub Repo Search
              </a>
            </h1>
            <p
              className="Logo_subTitle_1r70osc"
            >
              Browse repos via 
              <a
                href="https://developer.github.com/v3/"
              >
                the GitHub API
              </a>
            </p>
          </div>
        </div>
        <div
          className="Header_form_1m5hoq2 u-sm-flexGrow1"
        >
          <form
            className="SearchForm_e296pg"
            onSubmit={[Function]}
          >
            <label
              className="u-hiddenVisually"
              htmlFor="searchInput"
            >
              Search for a repo
            </label>
            <input
              className="SearchForm_input_12zt507"
              id="searchInput"
              onChange={[Function]}
              placeholder="Search for a repo"
              type="text"
              value=""
            />
            <div
              className="Dropdown-root"
            >
              <div
                aria-haspopup="listbox"
                className="Dropdown-control"
                onMouseDown={[Function]}
                onTouchEnd={[Function]}
              >
                <div
                  className="Dropdown-placeholder is-selected"
                >
                  JavaScript
                </div>
                <div
                  className="Dropdown-arrow-wrapper"
                >
                  <span
                    className="Dropdown-arrow"
                  />
                </div>
              </div>
            </div>
            <button
              className="SearchForm_btn_13vgzd5"
              type="submit"
            >
              <svg />
              <span
                className="u-hiddenVisually"
              >
                Submit
              </span>
            </button>
          </form>
        </div>
      </div>
    </div>
  </div>
  <div
    className="u-flexGrow1"
  >
    <p>
      testing
    </p>
  </div>
  <div
    className="Container_16pg94n-o_O-DefaultLayout_footer_15lfnry"
  >
    <div
      className="Container_inner_w9z6jj-o_O-Container_padding_1cv73u5"
    >
      <div
        className="u-flex u-flexWrap u-flexAlignItemsCenter u-flexJustifyBetween"
      >
        <p
          className="DefaultLayout_footerText_nl891q"
        >
          View the 
          <a
            href="https://github.com/rupeshdabbir/github-search-example"
          >
            source on GitHub
          </a>
        </p>
        <div
          className="u-flex u-flexAlignItemsCenter"
        >
          <p
            className="RateLimitContainer_label_18lea6b"
          >
            API requests
          </p>
          <div
            className="RateLimitContainer_xe3jre u-flex"
          >
            <div
              className="RateLimitContainer_item_ygrcmc"
            >
              <p
                className="RateLimit_f9e8qy u-flex"
                title="Limit resets at "
              >
                <span
                  className="RateLimit_label_1rcbo1g"
                >
                  Search
                  :
                </span>
                <span
                  className="RateLimit_low_1aa01pl"
                >
                  0
                </span>
              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
